guacamole-common-js/src/main/webapp/modules/KeyEventInterpreter.js

```diff
@@ -460,91 +253,85 @@ Guacamole.KeyEventInterpreter = function KeyEventInterpreter(batchSeperation, st
  */
-Guacamole.KeyEventInterpreter.KeyEvent = function KeyEvent(text, typed, timestamp) {
+Guacamole.KeyEventInterpreter.KeyDefinition = function KeyDefinition(template) {
+
+    // Use empty object by default
+    template = template || {};
 
     /**
-     * A human-readable representation of the event. If a printable character
-     * was directly typed, this will just be that character. Otherwise it will
-     * be a string describing the event.
-     *
-     * @type {!String}
+     * The X11 keysym of the key.
+     * @type {!number}
      */
-    this.text = text;
+    this.keysym = parseInt(template.keysym);
 
     /**
-     * True if this text of this event is exactly a typed character, or false
-     * otherwise.
-     *
-     * @type {!boolean}
+     * A human-readable name for the key.
+     * @type {!String}
      */
-    this.typed = typed;
+    this.name = template.name;
 
     /**
-     * The timestamp from the recording when this event occured. If a
-     * `startTimestamp` value was provided to the interpreter constructor, this
-     * will be relative to start of the recording. If not, it will be the raw
-     * timestamp from the key event.
-     *
-     * @type {!Number}
+     * The value which would be typed in a typical text editor, if any. If the
+     * key is not associated with any typeable value, this will be undefined.
+     * @type {String}
      */
-    this.timestamp = timestamp;
+    this.value = template.value;
 
 };
 
 /**
- * A series of intepreted key events, seperated by at least the configured
- * batchSeperation value from any other key events in the recording corresponding
- * to the interpreted key events. A batch will always consist of at least one key
- * event, and an associated simplified representation of the event(s).
+ * A granular description of an extracted key event, including a human-readable
+ * text representation of the event, whether the event is directly typed or not,
+ * and the timestamp when the event occured.
  *
  * @constructor
- * @param {!Guacamole.KeyEventInterpreter.KeyEvent[]} events
- *     The interpreted key events for this batch.
- *
- * @param {!String} simpleValue
- *     The simplified, human-readable value representing the key events for
- *     this batch.
+ * @param {Guacamole.KeyEventInterpreter.KeyDefinition|object} [template={}]
+ *     The object whose properties should be copied within the new
+ *     KeyDefinition.
  */
-Guacamole.KeyEventInterpreter.KeyEventBatch = function KeyEventBatch(events, simpleValue) {
+Guacamole.KeyEventInterpreter.KeyEvent = function KeyEvent(template) {
+
+    // Use empty object by default
+    template = template || {};
 
     /**
-     * All key events for this batch.
+     * The key definition for the pressed key.
      *
-     * @type {!Guacamole.KeyEventInterpreter.KeyEvent[]}
+     * @type {!Guacamole.KeyEventInterpreter.KeyDefinition}
      */
-    this.events = events || [];
+    this.definition = template.definition;
```

Comment:
`Guacamole.KeyEventInterpreter.KeyDefinition` is marked as `@private`, but this property is public.

Reply:
Ah, right you are. It shouldn't be private.